sys/net/crosslayer/ng_netreg/ng_netreg.c

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (C) 2015 Martine Lenders <mlenders@inf.fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file    ng_netreg.c
+ */
+
+#include <errno.h>
+
+#include "clist.h"
+#include "net/ng_netreg.h"
+#include "net/ng_nettype.h"
+#include "utlist.h"
+
+#define _INVALID_TYPE(type) (((type) <= NG_NETTYPE_UNDEF) || ((type) >= NG_NETTYPE_NUMOF))
+
+static ng_netreg_entry_t *netreg[NG_NETTYPE_NUMOF - 1]; /* leave out NG_NETTYPE_UNDEF */
```

Comment:
``/*pointers to entries of netreg_entries[], sorted by their type (ng_nettype_t). */``